Solutions/moritz_w/Lesson_2/lesson2.py

```diff
@@ -0,0 +1,65 @@
+import sys
+
+def main():
+    test_list = [0, 1, 2]
+    append_list = [3, 4]
+
+    try:
+        print("new_list1 = " + str(test_list) + " + 3")
+        new_list1 = test_list + 3
+        print("Result: " + str(new_list1))
+    except TypeError as e:
+        print(e)
+
+    try:
+        print("new_list2 = " + str(test_list) + " + " + str(append_list))
+        new_list2 = test_list + append_list
+        print("Result: " + str(new_list2))
+    except TypeError as e:
+        print(e)
+
+    print("Change type of a")
+    print("List:")
+    a = []
+    print(type(a))
+    print("Change type of a to bool")
+    a = True
+    print(type(a))
+
+    print("--- Mutable Types ---")
+    print("Test list: " + str(test_list))
+    print("append new list")
+    test_list += append_list
```

Comment:
Using += to mutate test_list is correct. But I want to point out here that it is not the same as test_list = test_list + append_list. Python list implements __add__ (for operator +) differently then __iadd__ (for operator +=). Difference is that using "+" new list is created that has both elements from test_list and append_list and this new list is than assigned to test_list. However using += actually modifies test_list itself.

You can also use here extend() list method as well.